argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,293 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DetectionCfg:
+    """Instantiates a DetectionCfg object for configuring a DetectionEvaluator.
+
+    Args:
+        sim_ths: The similarity thresholds for determining a true positive.
+        sim_fn_type: The type of similarity function to be used for calculating
+                     average precision.
+        n_rec_samples: Number of recall points to sample uniformly in [0, 1]. Default to 101 recall samples.
+        dcs: The weight vector for the detection composite score (DCS).
+        tp_threshold: Center distance threshold for the true positive metrics (in meters).
+        significant_digits: The precision for metrics.
+    """
+
+    sim_ths: List[float] = field(default_factory=lambda: [0.5, 1.0, 2.0, 4.0])
+    sim_fn_type: SimFnType = SimFnType.CENTER
+    n_rec_samples: int = 101
+    cds_weights: List[float] = field(default_factory=lambda: [3, 1, 1, 1])
+    tp_thresh: float = 2.0
+    significant_digits: int = 3
+    detection_classes: List[str] = field(default_factory=lambda: list(OBJ_CLASS_MAPPING_DICT.keys()))
+    save_figs: bool = False
+
+
+@dataclass
+class DetectionEvaluator:
+    """Instantiates a DetectionEvaluator object for evaluation.
+
+    Args:
+        dt_fpath: The path to the folder which contains the detections.
+        gt_fpath: The path to the folder which contains all the logs.
+        fig_fpath: The path to the folder which will contain the output figures.
+        dt_cfg: The detection configuration settings.
+    """
+
+    dt_fpath: Path = Path("detections")
+    gt_fpath: Path = Path("/data/argoverse")
+    fig_fpath: Path = Path("figs")
+    dt_cfg: DetectionCfg = DetectionCfg()
+
+    def evaluate(self) -> pd.DataFrame:
+        """Evaluate detection output and return metrics. The multiprocessing
+        library is used for parallel assignment between detections and ground truth
+        annotations.
+
+        Returns:
+            The evaluation metrics.
+        """
+        dt_fpaths = list(self.dt_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+        gt_fpaths = list(self.gt_fpath.glob("*/per_sweep_annotations_amodal/*.json"))
+
+        assert len(dt_fpaths) == len(gt_fpaths)
+        data: DefaultDict[str, np.ndarray] = defaultdict(list)
+        cls_to_ninst: DefaultDict[str, int] = defaultdict(int)
+
+        with Pool(os.cpu_count()) as p:
+            accum = p.map(self.accumulate, gt_fpaths)
+
+        for frame_stats, frame_cls_to_inst in accum:
+            for cls_name, cls_stats in frame_stats.items():
+                data[cls_name].append(cls_stats)
+            for cls_name, num_inst in frame_cls_to_inst.items():
+                cls_to_ninst[cls_name] += num_inst
+
+        data = defaultdict(np.ndarray, {k: np.vstack(v) for k, v in data.items()})
+
+        columns = ["AP", "ATE", "ASE", "AOE", "CDS"]
+
+        # Initialize metrics table [0, 1.0, 1.0, 1.0, 0] represents the default value for each column.
```

Comment:
what do the rows and columns represent? can we describe their representation in the module-level docstring?